src/aspire/classification/averager2d.py

```diff
@@ -395,78 +294,146 @@ def align(self, classes, reflections, basis_coefficients):
+            for x, y in tqdm(
+                zip(x_shifts, y_shifts),
+                total=len(x_shifts),
+                desc="\tmaximizing over shifts",
+                disable=len(x_shifts) == 1,
+                leave=False,
+            ):
+                shift = np.array([x, y], dtype=int)
+                logger.debug(f"Computing rotational alignment after shift ({x},{y}).")
+
+                # For each shift, the set of neighbor images is shifted.
+                #   This order is chosen because:
+                #   i) allows concatenation of shifts and rotation
+                #   operations after orientation estimation
+                #   ii) because generally the number of neighbors << the
+                #   number of test rotations.
+
+                # Skip zero shifting.
+                if np.any(shift != 0):
+                    # Note the base image[0] is never shifted.
+                    _images[1:] = original_images[1:].shift(shift).asnumpy()
+
+                # Convert to array of complex coef, implicit copy.
+                _coef = self.alignment_basis.evaluate_t(Image(_images))
+                _coef = xp.array(_coef.to_complex().asnumpy())
+
+                # Handle reflections
+                refl = reflections[k]
+                _coef[refl] = xp.conj(_coef[refl])
+
+                # Generate table of rotations for image 0.
+                # Note we invert the rotations later.
+                #   Applying rot to image 0
+                #   avoids rotating each member of the class
+                #   for the argmax alignment test.
+                base_img = _coef[0].reshape(self.alignment_basis.complex_count, 1)
+
+                # (cnt, n_transl) * (cnt, 1) -> (cnt, n_transl)
+                rot_base_imgs_conj = _rot_ops_conj * base_img.conj()
+
+                # Compute dot product of each base-neighbor pair.
+                #   The collection of dots is performed in bulk
+                #   as a large matmul.
+                # (n_nbor, cnt) @ (cnt, n_transl) = (n_nbor, n_transl)
+                dots = xp.real(_coef @ rot_base_imgs_conj)
+                idx = xp.argmax(dots, axis=1)
+                idx[0] = 0  # Force base image, just in case.
+
+                # Assign results for this class
+                _correlations[:] = xp.asnumpy(
+                    xp.take_along_axis(dots, idx.reshape(n_nbor, 1), axis=1).flatten()
+                )
+                # _correlations[0] = 1  # Force base correlation, just in case
+                #  Todo, discuss:
+                #      If we care to compute the an actual correlation?
+                #      This involves a lot of somewhat careful normalizing.
+                #      The MATLAB code did do something like this,
+                #      but only for diagnostic purposes.
+                #      In our code anything beyond a relative calculation
+                #      would currently be superflous.
```

Comment:
I think that makes sense, but let's rename it `dot_products` so that there's no confusion.

Reply:
Okay. I changed this code and the corresponding attribute from `correlations` ~> `dot_product`. This crept into some of the docs and a unit test, but was all string replacements in a single commit. No functional changes.